src/node_buffer.cc

```diff
@@ -693,16 +693,16 @@ void StringWrite(const FunctionCallbackInfo<Value>& args) {
   size_t max_length;
 
   CHECK_NOT_OOB(ParseArrayIndex(args[1], 0, &offset));
+  if (offset >= ts_obj_length)
+    return env->ThrowRangeError("Offset is out of bounds");
+
   CHECK_NOT_OOB(ParseArrayIndex(args[2], ts_obj_length - offset, &max_length));
 
   max_length = MIN(ts_obj_length - offset, max_length);
 
   if (max_length == 0)
     return args.GetReturnValue().Set(0);
 
-  if (offset >= ts_obj_length)
-    return env->ThrowRangeError("Offset is out of bounds");
-
```

Comment:
@trevnorris @indutny Tracked issue I got after upgrade to this point. Apparently this broke some of my code e.g. using `protocol-buffers` (and I'm sure some other binary encoders too) which, for certain data, are writing empty strings in the end of the buffer, which is safe operation (no-op) on its own and worked up until this PR without any extra checks.

I was going to prepare PR to fix this but sounds like it falls under

> There is one edge case with max_length equal to 0

and I'm wondering whether we consider this an expected behavior?

Minimal sample: `Buffer.alloc(0).write('')` - works in pre-6.4.0, throws now.